agilent_infiniium_scope.py

```diff
@@ -31,446 +31,354 @@
-        returnValue(resp)
+        yield dev.query('*CLS;*OPC?')
 
-    @setting(112, channel = 'i', scale = 'v', returns = ['v'])
-    def scale(self, c, channel, scale = None):
+    @setting(112, channel='i', scale='v', returns=['v'])
+    def scale(self, c, channel, scale=None):
         """Get or set the vertical scale of a channel in voltage per division.
         """
         dev = self.selectedDevice(c)
-        if scale is None:
-            resp = yield dev.query('CHAN%d:SCAL?' %channel)
-        else:
-            scale = format(scale,'E')
-            yield dev.write(('CHAN%d:SCAL '+scale) %channel)
-            resp = yield dev.query('CHAN%d:SCAL?' %channel)
+        if scale is not None:
+            yield dev.write('CHAN{}:SCAL {:E}'.format(channel, scale))
+        resp = yield dev.query('CHAN{}:SCAL?'.format(channel))
         scale = float(resp)
         returnValue(scale)
 
-    @setting(113, channel = 'i', factor = 'i', returns = ['s'])
-    def probeNONEXISTANT(self, c, channel, factor = None):
-        """Get or set the probe attenuation factor.
-        """
-        raise Exception('Not yet implemented')
-        probeFactors = [1,10,20,50,100,500,1000]
-        dev = self.selectedDevice(c)
-        chString = 'CH%d:' %channel
-        if factor is None:
-            resp = yield dev.query(chString+'PRO?')
-        elif factor in probeFactors:
-            yield dev.write(chString+'PRO %d' %factor)
-            resp = yield dev.query(chString+'PRO?')
-        else:
-            raise Exception('Probe attenuation factor not in '+str(probeFactors))
-        returnValue(resp)
-
-    @setting(114, channel = 'i', state = '?', returns = 's')
-    def channelOnOff(self, c, channel, state = None):
+    @setting(114, channel='i', state=['s', 'i'], returns='s')
+    def channelOnOff(self, c, channel, state=None):
         """Turn on or off a scope channel display.
+
         State must be in [0,1,'ON','OFF'].
         Channel must be int.
         If state is not specified, will return state of channel.
         """
         dev = self.selectedDevice(c)
-        if state is None:
-            resp = yield dev.query('CHAN%d:DISP?' %channel)
-        else:
-            if isinstance(state, int):
-                state = str(state)
-            elif isinstance(state, str):
+        if state is not None:
+            if isinstance(state, str):
                 state = state.upper()
-            else:
-                raise Exception('state must be int or string')
-            if state not in ['0','1','ON','OFF']:
-                raise Exception('state must be 0, 1, "ON", or "OFF"')
-            yield dev.write(('CHAN%d:DISP '+state) %channel)
-            resp = yield dev.query('CHAN%d:DISP?' %channel)
+            if state not in [0, 1, 'ON', 'OFF']:
+                raise Exception('state must be 0, 1, "ON", or "OFF". '
+                                'Got {}'.format(state))
+            if channel not in [1, 2, 3, 4]:
```

Comment:
There are a number of other settings that take `channel` as an argument that would benefit from the same check. It would be good to factor the check out into a separate method that can be reused. However, I'd suggest doing that in a separate commit since this one is already pretty big :-)